specification/azurefleet/AzureFleet.Management/virtualMachineScaleSet.tsp

```diff
@@ -0,0 +1,81 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using TypeSpec.OpenAPI;
+using Azure.ResourceManager;
+
+namespace Microsoft.AzureFleet;
+
+@doc("An AzureFleet's virtualMachineScaleSet")
+model VirtualMachineScaleSet {
```

Comment:
>model VirtualMachineScaleSet [](http://example.com/codeflow?start=0&length=28)

Use VirtualMachineScaleSet is ProxyResource<VirtualMachineScaleSetProperties> with a separate VirtualMachineScaleSetProperties definition that declares just the RP-specific properties. This automates the common ARM declarations.

Reply:
I tried making it a proxy resource but we don't need the extra fields which come with it. It is just a custom List operation.
![image](https://github.com/Azure/azure-rest-api-specs/assets/7030203/e93589e9-b999-4436-be24-007419b9029d)
Even Compute has the same for custom List Operation: https://learn.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-sets/list-skus?view=rest-compute-2024-03-01&tabs=HTTP

Reply:
Also if we did define VirtualMachineScaleSet as a proxy resource then we would have to implement get for a single VirtualMachineScaleSet as well. We don't want to follow that pattern.

Reply:
@mentat9 Please have a look.

Reply:
Per offline agreement, the full ProxyResource model will be added in the next API version. To support the the team's investment in this existing API version, ARM will sign off with just the list API, updated to match the ARM ProxyResource LIST format. The next API version will add the missing point GET API.